extension/js/common/platform/store/contact-store.ts

```diff
@@ -156,7 +166,24 @@ export class ContactStore extends AbstractStore {
   }
 
   /**
-   * used to update existing contact
+   * Used to update certain fields of existing contacts or create new contacts using the provided data.
+   * If an array of emails is provided, the update operation will be performed independently on each of them.
+   * Missing properties from update object will not be overwritten in the database,
+   * e.g. `{name} as ContactUpdate` will retain the `lastUse` value of the existing record,
+   * whereas `{lastUse:null} as ContactUpdate` will reset `lastUse` to null.
+   * The `pubkey` property will be used only to add or update a pubkey record by the pubkey fingerprint.
+   * Null value of `pubkey` won't affect any pubkey records.
+   * The `pubkeyLastCheck` property can be set to a non-null value only when `pubkey` specified
+   * and will be applied only to that specific pubkey record. Missing or null `pubkeyLastCheck` will
```

Comment:
There is a bit inconsistent behaviour. We can reset `name` and `lastUse` properties of the contact by providing null values in the `update` parameter, whereas `pubkeyLastCheck` will be ignored if `null` is provided. It's impossible to reset it. Should we do property existence check on it as with other properties or leave as it is?

Reply:
Name and lastUse should not be ever reset (no point setting name from something to null, and no point setting lastUse from something to null either). Thanks for noticing!